scripts/dts/gen_defines.py

```diff
@@ -493,15 +493,19 @@ def encode_zephyr_multi_level_irq(irq, irq_num):
         out_dt_define(macro, val)
 
 
-def write_compatibles(node):
+def write_compatibles(node, compat2vendor):
     # Writes a macro for each of the node's compatibles. We don't care
     # about whether edtlib / Zephyr's binding language recognizes
     # them. The compatibles the node provides are what is important.
 
-    for compat in node.compats:
+    for i, compat in enumerate(node.compats):
         out_dt_define(
             f"{node.z_path_id}_COMPAT_MATCHES_{str2ident(compat)}", 1)
 
+        if compat2vendor[compat]:
+            out_dt_define(f"{node.z_path_id}_COMPAT_VENDOR_IDX_{i}_EXISTS", 1)
+            out_dt_define(f"{node.z_path_id}_COMPAT_VENDOR_IDX_{i}",
+                          quote_str(compat2vendor[compat]))
```

Comment:
Use `node.edt.compat2vendor` instead of adding a separate argument all the way down.